### DISCUSSION TITLE
Ytuner behind NGINX

### DISCUSSION THREAD

**Discussion by Frederik-Lede** [coffeegreg/YTuner#26] [Q&A] [answered]
Do you have any documentation on how to put YTuner behind NGINX?

**Comment by coffeegreg** [comment]
Hi @Frederik-Lede
Unfortunately, I am not familiar with NGINX, but treat YTuner services as regular web services. 
Let's consider a simple configuration for a reverse proxy server such as nginx:
![reverse-proxy](https://github.com/user-attachments/assets/5e8964e3-0e8e-409f-9eb4-bee4d25736ce)

If YTuner is running on a host with an IPv4 address like 10.252.140.1 with standard ports (TCP 80-webservice + UDP 53-DNS service) and this YTuner host is reach out for external AVRs on 10.254.140.100 you should use `ActAsHost` parameter in `Configuration` section of `ytuner.ini` file and set it to 10.254.140.100. Maybe this case is similar and will be helpful : #25 and maybe @Larsvb0 could share his experiences in this area. Running YTuner with a Docker container doesn't matter here.
Let us know what you've managed to do and share your knowledge and experiences with us. Such information may be useful to someone in the future.

**Comment by Larsvb0** [comment]
Reverse proxy works.

On my router I have a DNS server running that runs all vtuner.denon.com to internal Synology server

On my Synology server I have this Reverse proxy:
http://denon.vtuner.com:80 --> http://internalip:8020

On the Nuc where Ytuner runs on I have it running on port 8020 using the configuration explained above.

**Comment by arunprasaddoss** [comment, marked as answer]
Hi @Frederik-Lede 

I am using my YTuner behind a nginx reverse proxy, see my configuration below:

## My Nginx Reverse Proxy Configuration using Docker

### Starting the Nginx Proxy Manager Docker Container

- Create a Docker Compose document for your environment

```
version: '3.8'
services:
  app:
    image: 'jc21/nginx-proxy-manager:latest'
    restart: unless-stopped
    ports:
      - '80:80'
      - '81:81'
      - '443:443' # Optional if needed
    volumes:
      - ./data:/data
      - ./letsencrypt:/etc/letsencrypt
```

- Run your docker container

```
docker-compose up -d
```
- Now, your Nginx Reverse Proxy is available, you can access the same from http://your-ip-address:81
```
Email:    admin@example.com
Password: changeme
```
- Change the password and get yourself used to the environment

### Setting up the YTuner for Reverse Proxy
- Download the YTuner from the [releases](https://github.com/coffeegreg/YTuner/releases/tag/1.2.3) page
- Update the ytuner.ini file to serve the YTuner on the different port:
```
;Web server TCP port. Don't change it if it's not necessary. ; Default = 80
WebServerPort=83
```
- Run your YTuner, please refere the [Installation instructions](https://github.com/coffeegreg/YTuner#Installation)

### Configuring the Nginx Reverse Proxy
- Now Login to your Niginx Proxy Manager and go to "Proxy Hosts"
- Click "Add Proxy Host"
- Up date your ip "Domain Names" field and put the YTuner port on the "Forward Port" and save
![image](https://github.com/user-attachments/assets/bc6533e9-e768-446c-9097-3d124b052bba)
- Now your YTuner service is avialable through the reverse proxy successfully
